Automattic/studio
# Studio: Delete all active demo sites

Related to https://github.com/Automattic/dotcom-forge/issues/6655
Fixes https://github.com/Automattic/dotcom-forge/issues/7081

## Proposed Changes

This PR fixes a bug where the user reaches the maximum number of active demo sites. However, if none of the created sites are linked to the demo sites, he can't remove any to continue creating demo sites. 

## Testing Instructions
Prerequisite: Apply D148749-code

1. Create some sites.
2. Edit `appdata-v1.json` files, and remove all snapshots
3. Restart the app
4. Go to settings, and ensure that you see a number of active demo sites like the below:
![2024-05-15T09:36:58,113953464+03:00](https://github.com/Automattic/studio/assets/497103/b45acd8b-7318-4579-8be6-603525163247)
5. Press the 3 dots menu, and then press the "Delete all demo sites" button.
6. Ensure that you see a message like below:
![2024-05-15T09:37:57,606601158+03:00](https://github.com/Automattic/studio/assets/497103/3f443656-0358-4d36-b3ab-d998c05af284)
7. Ensure that pressing cancel doesn't have any effect.
8. Repeat steps from 4-6.
9. On the dialog press "Delete all demo sites"
10. Notice that the sites are deleted, and you end up with 0 active demo sites.



## Pre-merge Checklist
